WebServerApiDesign.md

```diff
@@ -0,0 +1,221 @@
+# BurnFeed - A Decentralized Twitter-Like DApp
+
+## API
+
+# BurnFeed HTTP API
+
+**Base URL:** ``
+
+## General Authentication
+
+To access endpoints other than `Authenticate`, `Sign-in`, and `Register`, you need to include an `Authorization header` in your requests with a valid token. Tokens can be obtained through the `Authenticate` procedure.
+- **Possible request (with header):**
+```json
+curl -X POST -H "Content-Type: application/json" -H "Authorization: Bearer YOUR_TOKEN" -H "Nonce: YOUR_NONCE" -H "User-Wallet-Address: YOUR_WALLET_ADDRESS" -d '{
+    "type": "artifact",
+    "version": "0.1",
+    "subtype": "tweet",
+    "user": "0x95222290DD7278Aa3Ddd389Cc1E1d165CC4BAfe5",
+    "content": "This serves as an example of a tweet."
+}' http://localhost:8080/create-artifact
+```
+
+
+## Register User
+
+- **URL:** `/register`
+- **Method:** `POST`
+- **Purpose:** Register a new user with their Ethereum wallet address.
+- **Request Format:**
+```json
+  {
+      "walletAddress": "0xAbC1234def5678ghI901jklmNoPq234Rstuv567WXyZ"
+  }
+```
+- **Response Format:**
+```json
+Status message confirming user registration
+```
+
+## Sign In
+- **URL:** `/sign-in`
+- **Method:** `POST`
+- **Purpose:** Initiate the signing process by providing the wallet address.
+- **Request Format:**
+```json
+  {
+    "walletAddress": "0xAbC1234def5678ghI901jklmNoPq234Rstuv567WXyZ"
+  }
+```
+- **Response Format:**
+```json
+  {
+    "nonce": "123456789"
+  }
+```
+
+
+## Authenticate
+- **URL:** `/authenticate`
+- **Method:** `POST`
+- **Purpose:** Authenticate the user's signature using their nonce and grant a token.
+- **Request Format:**
+```json
+{
+    "walletAddress": "0xAbC1234def5678ghI901jklmNoPq234Rstuv567WXyZ",
+    "signedNonce": "0x..."
+}
+```
+- **Response Format:**
+```json
+{
+    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
+}
+```
+
+## Create artifact
+- **URL:** `/artifact`
```

Comment:
I would suggest to create one post API for reach *action* supported by the protocol. Then as many GET apis for fetching data for different view rendering.

Reply:
Regarding push: OK.

Regarding get: Ok, i'm waiting for AlZaeemOdion to include the user (wallet address) in the DB to query data (if someone wants to 'visit' a user's page).

Reply:
Added 1 common interface for putting files on IPFS. It is now a header-type separated `POST`. (Please see `curl_commands.md`)